### PR TITLE
Issue 171: Add support for custom include path

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: server/src/main/java/com/defold/extender/Extender.java [hunk 1]
```diff
@@ -445,8 +445,18 @@ else if(f.getName().endsWith(".aar")) {
...
+        includes.add( ExtenderUtil.getRelativePath(jobDirectory, extIncludeDir ) );
         includes.add( ExtenderUtil.getRelativePath(jobDirectory, uploadDirectory) );
+
+        // Add the other extensions include folders
+        for (File otherExtDir : this.getExtensionFolders()) {
+            File otherIncludeDir = new File(otherExtDir, "include");
+            if (extIncludeDir.equals(otherIncludeDir))
+                continue;
+            includes.add( ExtenderUtil.getRelativePath(jobDirectory, otherIncludeDir ) );
+        }
```
Comment (JCash) [inline server/src/main/java/com/defold/extender/Extender.java:458]: Now you can include from other extensions.

File: server/src/main/java/com/defold/extender/ExtenderUtil.java [hunk 2]
```diff
@@ -379,4 +380,9 @@ public static File getAndroidResourceFolder(File dir) {
         return null;
     }
 
+    public static boolean isChild(File parent, File child) {
+        Path parentPath = parent.toPath().normalize().toAbsolutePath();
+        Path childPath = child.toPath().normalize().toAbsolutePath();
+        return childPath.startsWith(parentPath);
+    }
```
Comment (JCash) [inline server/src/main/java/com/defold/extender/ExtenderUtil.java:387]: We don't want users to include files from outside their upload directory.

File: server/src/main/java/com/defold/extender/ExtensionManifestValidator.java [hunk 3]
```diff
@@ -64,6 +90,14 @@ void validate(String extensionName, Map<String, Object> context) throws Extender
...
                     break;
 
+                case "includes":
+                    if (!(v instanceof List)) {
+                        throw new ExtenderException(String.format("Error in '%s': The 'includes' must be a list of strings. Got %s: %s (type %s)", extensionName, k, v.toString(), v.getClass().getCanonicalName()));
+                    }
+
+                    validateIncludePaths(extensionName, extensionFolder, (List<String>) v);
+                    continue;
+
```
Comment (JCash) [inline server/src/main/java/com/defold/extender/ExtensionManifestValidator.java:100]: If they keyword isn't in this list, the build gets rejected.

**Review by britzl — APPROVED** [review]
Looks good to me!

**Pull request merged**